python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -625,7 +625,7 @@ class Join(IR):
     right_on: list[expr.NamedExpr]
     """List of expressions used as keys in the right frame."""
     options: tuple[
-        Literal["inner", "left", "full", "leftsemi", "leftanti", "cross"],
+        Literal["inner", "left", "right", "full", "leftsemi", "leftanti", "cross"],
```

Comment:
Added support for the new right join style, since it was not much more effort than raising as an unsupported type.